pycbc/inference/models/relbin.py

```diff
@@ -589,3 +607,184 @@ def extra_args_from_config(cp, section, skip_args=None, dtypes=None):
+        self.draw_ifos(self.ref_snr, **kwargs)
+
+    @property
+    def ref_snr(self):
+        if not hasattr(self, '_ref_snr'):
+            wfs = {ifo: (self.h00_sparse[ifo],
+                         self.h00_sparse[ifo]) for ifo in self.h00_sparse}
+            self._ref_snr = self.get_snr(wfs, reference=True)
+        return self._ref_snr
+
+    def get_snr(self, wfs, reference=False):
+        """ Return hp/hc maximized SNR time series
+        """
+        delta_t = 1.0 / self.sample_rate
+        snrs = {}
+        for ifo in wfs:
+            sdat = self.sdat[ifo]
+            dtc = self.tstart[ifo] - self.end_time[ifo]
+
+            if reference:
+                dtc -= self.ta[ifo]
+
+            snr = snr_predictor(self.fedges[ifo],
+                                dtc, delta_t,
+                                self.num_samples[ifo],
+                                wfs[ifo][0], wfs[ifo][1],
+                                self.h00_sparse[ifo],
+                                sdat['a0'], sdat['a1'],
+                                sdat['b0'], sdat['b1'])
+            snrs[ifo] = TimeSeries(snr, delta_t=delta_t,
+                                   epoch=self.tstart[ifo])
+        return snrs
+
+    def _loglr(self):
+        r"""Computes the log likelihood ratio,
+
+        .. math::
+
+            \log \mathcal{L}(\Theta) = \sum_i
+                \left<h_i(\Theta)|d_i\right> -
+                \frac{1}{2}\left<h_i(\Theta)|h_i(\Theta)\right>,
+
+        at the current parameter values :math:`\Theta`.
+
+        Returns
+        -------
+        float
+            The value of the log likelihood ratio.
+        """
+        # get model params
+        p = self.current_params
+        wfs = self.get_waveforms(p)
+        lik = self.likelihood_function
+        norm = 0.0
+        filt = 0j
+
+        self.snr_draw(self.get_snr(wfs))
+        p = self.current_params
+
+        for ifo in self.data:
+            freqs = self.fedges[ifo]
+            sdat = self.sdat[ifo]
+            h00 = self.h00_sparse[ifo]
+            end_time = self.end_time[ifo]
+            times = self.antenna_time[ifo]
+
+            hp, hc = wfs[ifo]
+            det = self.det[ifo]
+            fp, fc = det.antenna_pattern(p["ra"], p["dec"],
+                                         p["polarization"], times)
+            dt = det.time_delay_from_earth_center(p["ra"], p["dec"], times)
+            dtc = p["tc"] + dt - end_time
+
+            filter_i, norm_i = lik(freqs, fp, fc, dtc,
+                                   hp, hc, h00,
+                                   sdat['a0'], sdat['a1'],
+                                   sdat['b0'], sdat['b1'])
+            filt += filter_i
+            norm += norm_i
+        loglr = self.marginalize_loglr(filt, norm)
+        return loglr
+
+
+class RelativeTimeDom(RelativeTime):
+    """ Heterodyne likelihood optimized for time marginalization
+    """
```

Comment:
This needs some more documentation. What's the difference between this and `RelativeTime`? Also not crazy about the name... if `Domain` meant by `Dom`, just call it `RelativeTimeDomain`.